controllers/osbuild_controller_test.go

```diff
@@ -250,54 +250,80 @@ var _ = Describe("OSBuild Controller", func() {
 	Context("ComposeId is empty so create postCompose request ", func() {
 		BeforeEach(func() {
 			// given
-			osBuildRepository.EXPECT().PatchStatus(requestContext, osbuildEdgeContainerInstance, gomock.Any()).Return(nil)
-			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(osbuildEdgeContainerInstance, nil)
+			osBuildRepository.EXPECT().PatchStatus(requestContext, osbuildInstance, gomock.Any()).Return(nil)
+			osBuildRepository.EXPECT().Read(requestContext, instanceName, instanceNamespace).Return(osbuildInstance, nil)
 		})
 
-		It("should requeue for long duration if failed on postCompose with an error", func() {
+		DescribeTable("should requeue for long duration if failed on postCompose with an error", func(targetImageType osbuildv1alpha1.TargetImageType) {
 			// given
-			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).Return(nil, errFailed)
+			osbuildInstance.Spec.Details.TargetImage.TargetImageType = targetImageType
+			composerClient.EXPECT().PostComposeWithResponse(requestContext, gomock.Any()).DoAndReturn(
+				func(ctx context.Context, body composer.PostComposeJSONRequestBody, reqEditors ...interface{}) (*composer.PostComposeResponse, error) {
+					Expect(body.ImageRequest.ImageType == composer.ImageTypes(targetImageType))
```

Comment:
I know this works, but I think it'd be nicer to use the test syntax:
`Expect(body.ImageRequest.ImageType).To(Equal(composer.ImageTypes(targetImageType)))`
(in all `DoAndReturn` implementations)

Reply:
Done